chiselFrontend/src/main/scala/chisel3/core/MonoConnect.scala

```diff
@@ -136,6 +145,9 @@ object MonoConnect {
         //    CURRENT MOD   CURRENT MOD
         case (Output,       _) => issueConnect(sink, source)
         case (Internal,     _) => issueConnect(sink, source)
+        // Normally, this (an Input on the LHS of a connection) should generate an error,
+        //  but it's the way we indicate "DontCare"s
+        case (Input, Internal) if source.binding == DontCareBinding() => issueConnect(sink, source)
```

Comment:
Wait, does this make sense? You can assign a DontCare to an input? Why should a module be writing its inputs?

Reply:
I agree it looks screwy, but chisel3 currently generates "is invalid" for inputs and this is an attempt to allow generation of that explicitly.

Reply:
is that something we want to keep supporting? can we get rid of that?

Reply:
I don't feel qualified enough to comment, but we should settle this before merging.

Reply:
This sounds related to firrtl warnings for partially initialized bundles, especially if they'r ebidirectional